applications/DEMApplication/custom_strategies/strategies/explicit_solver_strategy.cpp

```diff
@@ -933,135 +940,160 @@ namespace Kratos {
 
         KRATOS_TRY
         ClearFEMForces();
-        ConditionsArrayType& pConditions = GetFemModelPart().GetCommunicator().LocalMesh().Conditions();
+        ConditionsArrayType& rConditions = GetFemModelPart().GetCommunicator().LocalMesh().Conditions();
         ProcessInfo& r_process_info = GetFemModelPart().GetProcessInfo();
         const ProcessInfo& r_const_process_info = GetFemModelPart().GetProcessInfo();
 
-        Vector rhs_cond;
-        Vector rhs_cond_elas;
-        std::vector<unsigned int> condition_partition;
-        OpenMPUtils::CreatePartition(mNumberOfThreads, pConditions.size(), condition_partition);
-        unsigned int index;
-
-        #pragma omp parallel for private (index, rhs_cond, rhs_cond_elas)
-        for (int k = 0; k < mNumberOfThreads; k++) {
-
-            ConditionsArrayType::iterator it_begin = pConditions.ptr_begin() + condition_partition[k];
-            ConditionsArrayType::iterator it_end = pConditions.ptr_begin() + condition_partition[k + 1];
 
-            for (ConditionsArrayType::iterator it = it_begin; it != it_end; ++it) { //each iteration refers to a different triangle or quadrilateral
+        struct my_tls {
+            Vector rhs_cond;
+            Vector rhs_cond_elas;
+        };
 
-                Condition::GeometryType& geom = it->GetGeometry();
+        //here the my_tls is constructed in place, which is the equivalent of "private" in OpenMP
+        block_for_each(rConditions, my_tls(), [&](Condition& rCondition, my_tls& rTLS){
```

Comment:
I am sorry but we need a test for this. I don't think the pressure/force on the walls is plotted in any of the existing tests currently. Without them, we can't know if we are breaking this part of the code.

Reply:
I think DEM_PRESSURE and the forces 
`                array_1d<double, 3>& node_rhs = geom[i].FastGetSolutionStepValue(CONTACT_FORCES);
                array_1d<double, 3>& node_rhs_elas = geom[i].FastGetSolutionStepValue(ELASTIC_FORCES);
                array_1d<double, 3>& node_rhs_tang = geom[i].FastGetSolutionStepValue(TANGENTIAL_ELASTIC_FORCES);`

are currently being tested in test_2D_contact.py and test_3D_contact.py? 
Although TANGENTIAL_ELASTIC_FORCES does not appear in the test.

Reply:
I thought they were with a single particle but apparently there are 8 particles.
@farrufat-cimne , we are running the tests in 1 thread currently (hardcoded 1). We should test them in several processors. I'll call you to speak about this.